app/(dashboard)/student.tsx

```diff
@@ -34,26 +45,98 @@ export default function StudentPage() {
             course {
               id
               title
+              startDate
+              startYear
+              yearDivision
             }
           }
         }
       }
     `,
     {}
   );
-  const courses = currentUserInfo.courseMemberships.map((x) => x.course);
+
+  const courses = currentUserInfo.courseMemberships.map(
+    (course) => course.course
+  );
+  const [filteredCourses, setFilteredCourses] = useState(courses);
+  const [notFocusesdcourses, setNotFocusesdcourses] = useState(courses);
+
+  const handleChange = (event: SelectChangeEvent<String>) => {
+    const selectedSort = event.target.value as String;
+    setSortBy(selectedSort);
+    setShowAllCourses(false);
+
+    // Update filteredCourses based on the selected sorting criteria
+    if (selectedSort === "title") {
+      // Sort courses alphabetically by title
+      setFilteredCourses(
+        [...courses].sort((a, b) => a.title.localeCompare(b.title))
+      );
+    } else if (selectedSort === "startYear") {
+      // Filter courses that match the currentYear
+      const filtered = courses.filter(
+        (course) => course.startYear === currentYear
+      );
+      filtered.sort((a, b) => b.startDate - a.startDate);
+      setFilteredCourses(filtered);
+      const notFocused = courses.filter(
+        (course) => course.startYear !== currentYear
+      );
+      notFocused.sort((a, b) => b.startDate - a.startDate);
+      setNotFocusesdcourses(notFocused);
+    } /* else if (selectedSort === "yearDivision") {
+      // Filter and sort courses with specific yearDivision criteria (e.g., FIRST_SEMESTER or SECOND_SEMESTER)
+      const filtered = courses.filter(
+        (course) =>
+          course.yearDivision === "FIRST_SEMESTER" &&
+          course.yearDivision !== null
+      );
+      filtered.sort((a, b) => b.startDate - a.startDate);
+      setFilteredCourses(filtered);
+
+      const notFocused = courses.filter(
+        (course) =>
+          course.yearDivision === "SECOND_SEMESTER" &&
+          course.yearDivision !== null
+      );
+      notFocused.sort((a, b) => b.startDate - a.startDate);
+      setNotFocusesdcourses(notFocused);
+    } */
+  };
+
+  const handleButton = () => {
+    setShowAllCourses(!showAllCourses);
+  };
 
   return (
     <main>
       <Typography variant="h1" gutterBottom>
         Dashboard
       </Typography>
+      {/* Sort by Selection */}
+      <Box sx={{ minWidth: 120, maxWidth: 150, float: "right" }}>
```

Comment:
Better not use float

Reply:
What would you suggest? Absolute right, was the only other thing I tried, but that is not useable in this use case.

Reply:
In most cases where you want to align something, flexbox does the trick